lib/Controller/SieveController.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @author Daniel Kesselberg <mail@danielkesselberg.de>
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Controller;
+
+use Horde\ManageSieve\Exception as ManagesieveException;
+use OCA\Mail\AppInfo\Application;
+use OCA\Mail\Db\MailAccountMapper;
+use OCA\Mail\Exception\ClientException;
+use OCA\Mail\Exception\CouldNotConnectException;
+use OCA\Mail\Service\AccountService;
+use OCA\Mail\Sieve\SieveClientFactory;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\IRequest;
+use OCP\Security\ICrypto;
+
+class SieveController extends Controller {
+
+	/** @var AccountService */
+	private $accountService;
+
+	/** @var MailAccountMapper */
+	private $mailAccountMapper;
+
+	/** @var SieveClientFactory */
+	private $sieveClientFactory;
+
+	/** @var string */
+	private $currentUserId;
+
+	/** @var ICrypto */
+	private $crypto;
+
+	/**
+	 * AccountsController constructor.
+	 *
+	 * @param IRequest $request
+	 * @param string $UserId
+	 * @param AccountService $accountService
+	 * @param MailAccountMapper $mailAccountMapper
+	 * @param SieveClientFactory $sieveClientFactory
+	 * @param ICrypto $crypto
+	 */
+	public function __construct(IRequest $request,
+								string $UserId,
```

Comment:
```suggestion
								string $userId,
```

Reply:
I think this one needs the weird first uppercase U because on the DI container the parameter is registered as `UserId`

Reply:
ah ok :)